modules/ixBidAdapter.js

```diff
@@ -2079,14 +2079,21 @@ function isValidAuctionConfig(config) {
 }
 
 /**
- * Adds device.w / device.h info
+ * Adds device info from ORTB2 data to the request
  * @param {object} r
+ * @param {object} ortb2Data
  * @returns object
  */
-export function addDeviceInfo(r) {
-  if (r.device == undefined) {
-    r.device = {};
+export function addDeviceInfo(r, ortb2Data) {
+  // add device object if not present
+  r.device = r.device || {};
+
+  // if present, merge device object from ortb2Data
+  if (ortb2Data?.device) {
+    mergeDeep(r.device, ortb2Data.device);
   }
+
+  // override screen width and height with window.screen values
```

Comment:
@mkendall07 do you know why this override exists?

Reply:
I guess this is the reason > https://github.com/prebid/Prebid.js/pull/10246#discussion_r1268581470

I haven't looked at the code but I guess core handles this now?

Reply:
https://github.com/prebid/Prebid.js/blob/dc6c0c35ead5f44f3fc35852a84956a6c9036bed/src/fpd/enrichment.js#L100

Reply:
slightly different logic it seem https://stackoverflow.com/questions/16162639/difference-between-screen-and-window-property

Reply:
Hi @patmmccann!

> do you know why this override exists?

I've noticed that this approach is common in many adapters. I believe the reason lies in the OpenRTB specification, where `w` and `h` are defined as the physical height and width <ins>_of the screen_</ins> in pixels. In contrast, the default implementation returns the viewport size.

If you'd like, I can create an issue to discuss this topic further. It might be something that should be addressed at the Prebid.js FPD level.

Reply:
Please do, thanks!